File: src/main/java/com/galaxy/Restaurantinformationsystem/repository/ReviewRepository.java
package com.galaxy.Restaurantinformationsystem.repository;

import com.galaxy.Restaurantinformationsystem.entity.ReviewEntity;
import com.galaxy.Restaurantinformationsystem.entity.StoreEntity;
import com.galaxy.Restaurantinformationsystem.entity.UserEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface ReviewRepository extends JpaRepository<ReviewEntity, Long> {
    void deleteByStoreEntity(StoreEntity storeEntity);

    List<ReviewEntity> findByUserEntity_Id(Long id);

    void deleteByUserEntity(UserEntity userEntity);

}

File: src/main/java/com/galaxy/Restaurantinformationsystem/repository/ReviewRepository.kt
package com.galaxy.Restaurantinformationsystem.repository

import com.galaxy.Restaurantinformationsystem.entity.ReviewEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

/**
 * 음식점 후기 Repository 클래스
 *
 * @author 김남주
 */
@Repository
interface ReviewRepository : JpaRepository<ReviewEntity, Long> {
    fun findByAuthorId(authorId: Long): List<ReviewEntity>
    fun findByStoreId(storeId: Long): List<ReviewEntity>
}
